decentraland/unity-renderer
# feat: better scene metrics (memory)

## What does this PR change?

This PR implements limits score counting for memory. The memory we are counting per scene covers:

- Textures
- Meshes
- AnimationClips
- AudioClips

For now, the counting is enabled for GLTF loading only. Convering asset bundle loading requires to bundle meta-data to each AB package and will be covered in other PR.

The memory metrics are counted only, not limited or enforced in any way (yet).

The scoring formulas aren't exact. To deal with the lack of accuracy in a safe way, they are designed to overshoot the actual memory footprint. This is covered by tests.

## How to test the changes?

https://play.decentraland.zone/?renderer-branch=feat/better-scene-metrics

As this PR doesn't add visible functionality, you should at least see that the existing limit counting mechanics keep working.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
